maven-resolver/src/main/java/org/wildfly/channel/maven/VersionResolverFactory.java

```diff
@@ -125,6 +127,38 @@ public File resolveArtifact(String groupId, String artifactId, String extension,
             }
             return result.getArtifact().getFile();
         }
+
+        @Override
+        public List<File> resolveArtifacts(List<? extends ArtifactCoordinate> coordinates) throws UnresolvedMavenArtifactException {
+            requireNonNull(coordinates);
+
+            List<ArtifactRequest> requests = new ArrayList<>();
+            for (ArtifactCoordinate coord : coordinates) {
+                requireNonNull(coord.getGroupId());
```

Comment:
you should not need it as the `ArtifactCoordinate` ctor wil already have these `requireNonNull`